polaris-core/src/main/java/org/apache/polaris/core/persistence/MetaStoreManagerFactory.java

```diff
@@ -33,7 +32,7 @@
  * configuration
  */
 @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
-public interface MetaStoreManagerFactory extends Discoverable {
```

Comment:
Interesting, isn't this going to disturb Dropwizard's so-called ["polymorphic configuration"](https://www.dropwizard.io/en/release-3.0.x/manual/configuration.html#polymorphic-configuration)?

It seems we rely heavily on that feature though:

https://github.com/apache/polaris/blob/7b2cca4add23e7be369971774a55ce3dae7cca14/polaris-service/src/main/resources/META-INF/services/io.dropwizard.jackson.Discoverable

Reply:
Indeed I think we cannot safely remove `Discoverable` here

Reply:
well, it is loaded only in one place on startup (as far as I can tell) and I did verify that call path manually. Jackson was able to deserialize config and the store impl. correctly.

Reply:
FWIW, listing the factory class in `META-INF/services/io.dropwizard.jackson.Discoverable` _is_ important, but implementing `Discoverable` is not.

Reply:
the other relevant service declaration is https://github.com/apache/polaris/blob/8cb6b44bf57dc597dab612d109d3eb534aef5715/polaris-service/src/main/resources/META-INF/services/org.apache.polaris.core.persistence.MetaStoreManagerFactory#L21

Reply:
@eric-maynard : what other test can validate correct operation without implementing `Discoverable`?

Reply:
My understanding is that removing `Discoverable` from this will break any implementations that want to supply their own configs (I think that's the polymorphic configuration mentioned above?). 

If you can provide such an implementation in a test and the configs still get picked up, I think we would be good to merge

Reply:
My understanding is that removing Discoverable from this will break any implementations that want to supply their own configs (I think that's the polymorphic configuration mentioned above?).

If you can provide such an implementation in a test and the configs still get picked up, I think we would be good to merge

Reply:
@eric-maynard : I've added `PolarisApplicationConfigurationTest` in the latest commit. Please have a look.

Reply:
@eric-maynard : WDYT?

Reply:
The changes look good to me, but the test doesn't include any configs specific to that implementation.

Reply:
The new test in this PR demonstrates config polymorphism between two different implementations (coming from different modules/jars). @eric-maynard : could you give a bit more details on what you'd like to be added to tests?

Reply:
I think he meant to enhance the test with some fields populated through configuration:

```java
  @Nested
  class EclipseLinkMetastore {
    private final DropwizardAppExtension<PolarisApplicationConfig> app =
        new DropwizardAppExtension<>(
            PolarisApplication.class,
            CONFIG_PATH,
            RANDOM_APP_PORT,
            RANDOM_ADMIN_PORT,
            ConfigOverride.config("metaStoreManager.type", "eclipse-link"),
            ConfigOverride.config("metaStoreManager.persistence-unit", "test"),
            ConfigOverride.config("metaStoreManager.conf-file", "/path/to/persistence.xml"));

    @Test
    void testMetastoreType() {
      assertThat(app.getConfiguration().getMetaStoreManagerFactory())
          .asInstanceOf(type(EclipseLinkPolarisMetaStoreManagerFactory.class))
          .extracting("persistenceUnitName", "confFile")
          .containsExactly("test", "/path/to/persistence.xml");
    }
  }
```

(you need to switch the eclipse-link dependency from `testRuntimeOnly` to `testImplementation`)

Reply:
ok, will do

Reply:
added